alacritty/src/renderer/rects.rs

```diff
@@ -285,28 +287,51 @@ impl RectRenderer {
             gl::BindBuffer(gl::ARRAY_BUFFER, self.vbo);
 
             gl::UseProgram(self.program.id());
+            self.program.update_uniforms(size_info, metrics);
         }
 
         let half_width = size_info.width() / 2.;
         let half_height = size_info.height() / 2.;
 
         // Build rect vertices vector.
-        self.vertices.clear();
+        self.rect_vertices.clear();
+        self.curl_vertices.clear();
         for rect in &rects {
-            self.add_rect(half_width, half_height, rect);
+            if rect.is_undercurl {
+                Self::add_rect(&mut self.curl_vertices, half_width, half_height, rect);
+            } else {
```

Comment:
Curious what this would look like with dotted and dashed added too. Would we add that point have 4 different Vecs?

Reply:
No, it just depends how you'd draw them. I think you can just push more rects for them drawn with `rect_verticies`. It just not that you can build up a good function to draw dashed and dotted undeline.